liblll/CompilerState.cpp

```diff
@@ -49,7 +49,8 @@ void CompilerState::populateStandard()
 	"(def 'allgas (- (gas) 21))"
 	"(def 'send (to value) (call allgas to value 0 0 0 0))"
 	"(def 'send (gaslimit to value) (call gaslimit to value 0 0 0 0))"
-	"(def 'msg (gaslimit to value data datasize outsize) { (set x outsize) (set y (alloc @32)) (call gaslimit to value data datasize @0 @32) @0 })"
+	// NOTE: in this macro, memory location 0 is set in order to force msize to be at least 32 bytes.
+	"(def 'msg (gaslimit to value data datasize outsize) { [0]:0 [0]:(msize) (call gaslimit to value data datasize @0 outsize) (mload @0) })"
```

Comment:
Since this changes behaviour anyway, would it make sense returning `@0` only so the caller knows the offset and can load from there (up to `outsize` length)?